pgrx/src/varlena.rs

```diff
@@ -12,17 +12,10 @@ Use of this source code is governed by the MIT license that can be found in the
 use crate::{pg_sys, PgBox};
 use core::{slice, str};
 
-/// # Safety
-///
-/// The caller asserts the specified `ptr` really is a non-null, palloc'd [`pg_sys::varlena`] pointer
-#[inline(always)]
-pub unsafe fn set_varsize_4b(ptr: *mut pg_sys::varlena, len: i32) {
-    // #define SET_VARSIZE_4B(PTR,len) \
-    // 	(((varattrib_4b *) (PTR))->va_4byte.va_header = (((uint32) (len)) << 2))
-
-    // SAFETY:  A varlena can be safely cast to a varattrib_4b
-    (*ptr.cast::<pg_sys::varattrib_4b>()).va_4byte.as_mut().va_header = (len as u32) << 2;
-}
+#[cfg(target_endian = "big")]
+pub use varlena_bigendian::*;
+#[cfg(target_endian = "little")]
+pub use varlena_littleendian::*;
```

Comment:
Given what you said
```suggestion
#[cfg(any(feature = "greenplum7", target_endian = "big"))]
pub use varlena_bigendian::*;
#[cfg(not(any(feature = "greenplum7", target_endian = "big")))]
pub use varlena_littleendian::*;
```

Reply:
Just to be clear, @workingjubilee's suggestion here is just for demonstration purposes.  We are not in any position to invent a `greenplum7` feature yet.  Such a thing will require a significant amount of planning, design, and understanding on our part, along with the ability to test it... and test it, apparently, on disparate CPU architectures.

Reply:
My goal is to add greenplum support, but this PR is not related to greenplum. and this PR did not add the `greenplum7` feature. If this PR gets merged, I will add gp7 support based on this PR.